nav2_planner/src/planner_server.cpp

```diff
@@ -510,7 +516,52 @@ PlannerServer::publishPlan(const nav_msgs::msg::Path & path)
   }
 }
 
-}  // namespace nav2_planner
+void PlannerServer::isPathValid(
+  const std::shared_ptr<nav2_msgs::srv::IsPathValid::Request> request,
+  std::shared_ptr<nav2_msgs::srv::IsPathValid::Response> response)
+{
+  unsigned int mx = 0;
+  unsigned int my = 0;
+
+  response->is_valid = true;
+
+  if (request->path.poses.empty() ) {
+    response->is_valid = false;
+  } else {
+    geometry_msgs::msg::PoseStamped current_pose;
+    unsigned int closest_pose_index = 0;
+    if (costmap_ros_->getRobotPose(current_pose)) {
+      tf2::Vector3 curr_position(current_pose.pose.position.x,
```

Comment:
You can use https://github.com/ros-planning/navigation2/blob/main/nav2_util/include/nav2_util/geometry_utils.hpp#L82 instead so we're not working with `tf2` types that we don't have control over. Use `PoseStamped` instead and use the nav2_util function to get the distance. Then you don't need to convert the types.

Reply:
Done.